torchgeo/models/neuralgcm.py

```diff
@@ -0,0 +1,116 @@
+}
+
+
+class NeuralGCM_Weights(WeightsEnum):  # type: ignore[misc]
+    """NeuralGCM weights.
+
+    If you use this model in your research, please cite the following paper:
+
+    * https://www.nature.com/articles/s41586-024-07744-y
+
+    .. versionadded:: 0.8
+    """
+
+    DETERMINISTIC_0_7_DEG = Weights(
+        url='gs://neuralgcm/models/v1/deterministic_0_7_deg.pkl',
+        transforms=_neuralgcm_transforms,
+        meta=_neuralgcm_meta | {'model': 'NeuralGCM07Deterministic', 'resolution': 0.7},
+    )
+
+    DETERMINISTIC_1_4_DEG = Weights(
+        url='gs://neuralgcm/models/v1/deterministic_1_4_deg.pkl',
+        transforms=_neuralgcm_transforms,
+        meta=_neuralgcm_meta | {'model': 'NeuralGCM14Deterministic', 'resolution': 1.4},
+    )
+
+    DETERMINISTIC_2_8_DEG = Weights(
+        url='gs://neuralgcm/models/v1/deterministic_2_8_deg.pkl',
+        transforms=_neuralgcm_transforms,
+        meta=_neuralgcm_meta | {'model': 'NeuralGCM28Deterministic', 'resolution': 2.8},
+    )
+
+    STOCHASTIC_1_4_DEG = Weights(
+        url='gs://neuralgcm/models/v1/stochastic_1_4_deg.pkl',
+        transforms=_neuralgcm_transforms,
+        meta=_neuralgcm_meta | {'model': 'NeuralGCM14Stochastic', 'resolution': 1.4},
+    )
+
+    STOCHASTIC_PRECIP_2_8_DEG = Weights(
+        url='gs://neuralgcm/models/v1_precip/stochastic_precip_2_8_deg.pkl',
+        transforms=_neuralgcm_transforms,
+        meta=_neuralgcm_meta
+        | {'model': 'NeuralGCM28StochasticPrecip', 'resolution': 2.8},
+    )
+
+    STOCHASTIC_EVAP_2_8_DEG = Weights(
+        url='gs://neuralgcm/models/v1_precip/stochastic_evap_2_8_deg.pkl',
+        transforms=_neuralgcm_transforms,
+        meta=_neuralgcm_meta
+        | {'model': 'NeuralGCM28StochasticEvap', 'resolution': 2.8},
+    )
+
+
+def neuralgcm(
+    weights: NeuralGCM_Weights | None = None, *args: Any, **kwargs: Any
+) -> nn.Module:
+    """NeuralGCM PressureLevelModel.
+
+    If you use this model in your research, please cite the following paper:
+
+    * https://www.nature.com/articles/s41586-024-07744-y
+
+    This model requires the following additional library to be installed:
+
+    * `neuralgcm <https://pypi.org/project/neuralgcm/>`_ to load the models.
+
+    .. versionadded:: 0.8
+
+    Args:
+        weights: Pre-trained model weights to use.
+        *args: Additional arguments to pass to ``neuralgcm.PressureLevelModel``
+        **kwargs: Additional keyword arguments to pass to ``neuralgcm.PressureLevelModel``
+
+    Returns:
+        A NeuralGCM PressureLevelModel.
+    """
+    lazy_import('neuralgcm')
+    import neuralgcm
+
+    if weights is None:
+        msg = 'NeuralGCM requires pre-trained weights. Please specify a weights enum.'
+        raise ValueError(msg)
+
+    # Load checkpoint from Google Cloud Storage
+    gcs = gcsfs.GCSFileSystem(token='anon')
+    with gcs.open(weights.url, 'rb') as f:
+        checkpoint = pickle.load(f)
```

Comment:
Can we add this as a function to torchgeo/models/utils.py? Would be good to be able to reuse this.